src/main/java/janggi/game/GameController.java

```diff
@@ -0,0 +1,46 @@
+package janggi.game;
+
+import janggi.board.Position;
+import janggi.db.BoardStatus;
+import janggi.piece.Piece;
+import janggi.team.Team;
+import janggi.team.Teams;
+import janggi.view.Input;
+import janggi.view.Output;
+
+public class GameController {
+    private final Output output;
+    private final GameProcessManager gameProcessManager;
+    private final GameRequestValidator gameRequestValidator;
+
+    public GameController(BoardStatus boardStatus) {
+        Input input = new Input();
+        this.output = new Output();
+        this.gameProcessManager = new GameProcessManager(input, boardStatus);
+        this.gameRequestValidator = new GameRequestValidator(input);
+    }
+
+    public void run() {
+        Teams teams = gameProcessManager.initializeTeams();
+        Team teamCho = teams.getTeamCho();
+        Team teamHan = teams.getTeamHan();
+        Team oldTeam = teamHan;
+        do {
+            output.printBoard(teamHan, teamCho);
+            Team currentTeam = teams.switchTurn(oldTeam);
+            oldTeam = currentTeam;
+
+            Piece currentPiece = gameRequestValidator.requestAndValidateStartPoint(currentTeam);
+            Position destination = gameRequestValidator.requestAndValidateDestination(currentTeam, currentPiece);
+            String pieceName = currentPiece.getName();
+            Position currentPosition = currentPiece.getPosition();
+
+            currentTeam.move(pieceName, currentPosition, destination);
+            Team opponentTeam = teams.checkOpponent(currentTeam);
+            opponentTeam.updateStatusIfCaught(destination);
+        } while (gameProcessManager.isContinue(teamHan, teamCho));
+
+        output.printTeamScore(teamCho.checkTeamScore(), teamHan.checkTeamScore());
+        gameProcessManager.saveTeams(teamCho, teamHan);
```

Comment:
https://github.com/woowacourse/java-janggi/pull/113#discussion_r2020189148 코멘트에서 아래와 같은 변경을 기대했답니다.

```suggestion
        Teams teams = gameProcessManager.initializeTeams();
        do {
            output.printBoard(teams);
            Team currentTeam = teams.getCurrentTeam();

            Piece currentPiece = gameRequestValidator.requestAndValidateStartPoint(currentTeam);
            Position destination = gameRequestValidator.requestAndValidateDestination(currentTeam, currentPiece);
            String pieceName = currentPiece.getName();
            Position currentPosition = currentPiece.getPosition();

            currentTeam.move(pieceName, currentPosition, destination);
            Team opponentTeam = teams.checkOpponent(currentTeam);
            opponentTeam.updateStatusIfCaught(destination);

            teams.switchTurn();
        } while (gameProcessManager.isContinue(teams));

        output.printTeamScore(teams);
        gameProcessManager.saveTeams(teams);
```

Reply:
오 네, `Teams` 클래스를 좀 더 활용하는 방식으로 `GameController` 클래스의 구조를 개선 해볼 수 있었네요. 해당 부분을 반영해서 좀 더 리팩토링 해보겠습니다!